server/routes/userSearch.js

```diff
@@ -0,0 +1,58 @@
+const express = require('express')
+const router = express.Router()
+
+const { PrismaClient } = require('../generated/prisma');
+const prisma = new PrismaClient()
+
+const { isAuthenticated } = require('../middleware/CheckAutheticated')
+
+
+//user search endpoint
+router.get('/search/users', isAuthenticated, async (req, res) => {
+
+    const { searchQuery } = req.query;
+
+    /*note: usernames should be one word, so if the search query had a space, that means the user is searching for a first and last name. If there is a second keyword, search by matching
```

Comment:
Commenting from a UX perspective, but might be better to search for both usernames and names that match or allowing users to specify a filter -- username, name, last name, location, etc